### DISCUSSION TITLE
Contribute types for the Vue plugin to Definitely Typed

### DISCUSSION THREAD

**Discussion by simensen** [tighten/ziggy#602] [Ideas]
What is the best way to get involved with the Definitely Typed work currently attributed to @benallfree? Is there a workflow/contribution pipeline for that or would it be best to reach out to @benallfree directly?

I've started using TypeScript with my new Vue app and finding eslint has trouble with the Vue plugin.

**Comment by benallfree** [comment]
Hey @simensen I'm here 👍 

You can also head over to DefinitelyTyped and submit a PR. I think the Ziggy team decided to keep it as a separate package.

**Reply by simensen** [reply]
@benallfree OK! I wasn't aware how Definitely Typed worked at all, and wasn't sure if you were active here or not. :) If submitting a PR to Definitely Typed is the best way to do this, I'll see if I can do that over there. I'm new to creating types for existing packages so I'm not 100% sure how it is supposed to work, but now I know where to put the work if I start it. :)

If I have questions or need some direction, what's the best way for me to communicate with you or someone else at Definitely Typed?

**Reply by benallfree** [reply]
@simensen Here is how to get started contributing to DT: https://github.com/DefinitelyTyped/DefinitelyTyped#how-can-i-contribute

Basically, you'll fork the DT repo, make changes to the ziggy types project, then submit a PR to DT. They have a lot of automated checks, but once everything passes, it will make it's way to me for review because I am listed as one of the maintainers. I'll review/approve the changes just like any normal PR. Once approved by me, the automated DT system will take over again and publish a new version of the ziggy types package.

I hope that helps!

Contributing to DT is kind of fun because it's a very advanced project with lots of contributors and lots of automation.
